rest/config.go

```diff
@@ -696,6 +696,15 @@ func (dbConfig *DbConfig) validateVersion(isEnterpriseEdition bool) error {
 		}
 	}
 
+	if validateOIDCConfig && dbConfig.OIDCConfig != nil {
+		for name, provider := range dbConfig.OIDCConfig.Providers {
+			_, _, err := provider.DiscoverConfig(context.TODO())
+			if err != nil {
+				multiError = multiError.Append(fmt.Errorf("failed to validate OIDC configuration for %s: %w", name, err))
```

Comment:
It will get logged. We have a `RedactableError` type but I'm not sure it plays nicely with `MultiError` OR `%w` :/

In similar cases we've just tagged and redacted the value inside the error, even if it's going to be displayed to the user of the API (e.g. in `CouchbaseBucketGoCB) Touch`)

```
2022-04-22T15:18:04.387+01:00 [INF] HTTP: c:#001 POST /db2/_config (as <ud>Administrator</ud> as ADMIN)
2022-04-22T15:18:05.435+01:00 [TRC] Config+: merging upserted config into bucket config -- rest.(*handler).handlePutDbConfig.func1() at admin_api.go:529
2022-04-22T15:18:08.576+01:00 [WRN] Couldn't update config for database - rolling back: 400 1 errors:
failed to validate OIDC configuration for fooprovider: Get "https://foo.bbrks.me/.well-known/openid-configuration": x509: “TRAEFIK DEFAULT CERT” certificate is not trusted -- rest.(*handler).handlePutDbConfig() at admin_api.go:571
2022-04-22T15:18:10.712+01:00 [TRC] Config+: Got config for bucket "b2" with cas 1650637004435881984 -- rest.(*ServerContext).fetchDatabase() at config.go:1154
2022-04-22T15:18:10.712+01:00 [DBG] Config+: Database "db2" bucket "b2" config has not changed since last update
2022-04-22T15:18:10.712+01:00 [INF] HTTP: c:#001 #001:     --> 400 1 errors:
failed to validate OIDC configuration for fooprovider: Get "https://foo.bbrks.me/.well-known/openid-configuration": x509: “TRAEFIK DEFAULT CERT” certificate is not trusted  (6327.9 ms)
```

Reply:
Discussed this on Slack, and even if we redact the provider name (`%s`), we can't (easily) redact the wrapped error (`%w`) which can contain the issuer as in the above example. However, this is config provided by the SG administrator (so would be in a sgcollect), so not strictly user data, so not redacting is likely fine here.